Pyfeast Expansion

barbacbd opened the pull request:
** Added all of the algorithms in the Feature Selection 

** Added an install bash script for PYFeast to build Feast and MIToolbox

bieito98 committed:
Allow lib MIToolbox in a sibling directory

Craigacp committed:
Merge pull request #12 from bieito98/master

Improve portability

Craigacp committed:
Delete testFSToolbox.m

Removing an old test file which shouldn't have been added.

barbacbd committed:
Starting to add each of the functions from fstoolbox library (FEAST).

barbacbd committed:
Finished adding the rest of the functions as class extensions. Still need to implement the call function for each.

barbacbd committed:
Updated feast information.

barbacbd committed:
Added build files for building my work. This is a script to pre build all dependencies for the project.

barbacbd committed:
Added data to the build script.

barbacbd committed:
Have multiple functions working (I believe).

barbacbd committed:
Added all functions in. Looks like the normal functions are working, but I am unsure on the discretized outputs.

barbacbd committed:
Simplifying the comments and structure at the top of the file for loading information.

barbacbd committed:
Merge branch 'master' into pyfeast_v2

barbacbd commented:
Did not get rid of the original pyfeast information. This can be deprecated and then removed .

barbacbd committed:
Testing, forgot to add the "." to the sudo install

Craigacp commented on python/feast/feast.py:
```diff
@@ -0,0 +1,589 @@
+
+
+
+def FeatureSelectionCommon(data_set, labels, num_features_to_select, *args, **kwargs):
```
Can this be made module private (i.e., start with an `_`)? I worry users might get weird error messages if they accidentally supply arguments which are unexpected for the underlying C function.

barbacbd replied:
oh absolutely I can make this one private. All of them will be made private and I can change the target as well as add some additional information to the setup. I will add a python readme too. I wrote a bash script that will allow the user to run everything from a single executable file to make the setup a bit easier.

Craigacp replied:
Thanks. For Python it would probably be nicer to figure out how to package the native libraries into the python library install as that's how most of the other Python libraries with native extensions work, but I can look at doing that separately.

Craigacp commented on python/feast/feast.py:
```diff
@@ -0,0 +1,589 @@
+    return selected_features, feature_scores
+
+
+def discFeatureSelectionCommon(data_set, labels, num_features_to_select, *args, **kwargs):
```
Same comment about making it module private.

Craigacp commented on python/feast/feast.py:
```diff
@@ -0,0 +1,589 @@
+    return selected_features, feature_scores
+
+
+def WeightedFeatureSelectionCommon(data_set, labels, weights, num_features_to_select):
```
Same comment about making this module private.

Craigacp reviewed (changes requested):
Thanks, this is really great. Would you mind targeting this PR at the `devel` branch? That way it'll be a little clearer which bits you're adding vs things coming from `devel` that haven't made it into `master`. Once the PR has merged I'll bounce things over into `master` and tag a fresh release.

Do you want to add a copyright statement to your contribution? I'd like to add the BSD license header to the Python file to make it clear it's under the same license as the rest of FEAST in the released version.

barbacbd committed:
Added README and made private settings

** Set several functions to internal or private use only for python to avoid
accidental calls to public functions or use with import *

** Added a README with an example of how to use the py feast extension.

** setup.py will read the license file from the main project directory.

** Removed old python feast files to avoid confusion. Since this isnt a true pypi package then we
can remove without deprecation since it is all built locally.

** Changed name in README to avoid confusion.

** Removed suspect $(libs) from makefile

Craigacp commented on Makefile:
```diff
@@ -45,8 +45,8 @@ JNI_INCLUDES = -I/usr/lib/jvm/java-8-openjdk-amd64/include/ -I/usr/lib/jvm/java-
-libFSToolbox.so : libMIToolbox.so $(objects)
-	$(LINKER) $(CFLAGS) -shared -o libFSToolbox.so $(objects) -lm -lMIToolbox
+libFSToolbox.so : $(objects)
+	$(LINKER) $(CFLAGS) -shared -o libFSToolbox.so $(objects) $(LIBS) -lm -lMIToolbox
```
My Make syntax is rather rusty, but doesn't this change stop the compilation of `libFSToolbox.so` from depending on the compilation of `libMIToolbox`?

barbacbd replied:
That may have snuck in a rebase that squashed something that you did previously.

Craigacp commented on python/README.md:
```diff
@@ -0,0 +1,55 @@
+# PyFeast
```
Could you change this to FEAST for Python, or something similar? There is already an [old binding](https://github.com/EESI/PyFeast) called PyFeast, but it doesn't share any code with this one, and the old PyFeast has been broken for a long time.

Craigacp reviewed (changes requested):
I think the only thing that's missing is adding the BSD license header to the new python file.

barbacbd committed:
** Added the license information and copyright data from feast to the pyfeast file.

barbacbd commented:
I believe that all required license information is added to the python files

Craigacp reviewed (approved):
Thanks for all your work on this. I'll work up some docs changes and then merge `devel` into `master` and tag a new release.
